packages/server/src/app/db/works/works.service.ts

```diff
@@ -154,38 +154,81 @@ export class WorksService {
         return await this.workModel.findOne({"_id": workId, "author": user.sub}).where("audit.isDeleted", false);
     }
 
+    /**
+     * Finds the first six matches given the provided search parameters.
+     * For use with the initial page of search results.
+     * 
+     * @param query The relevant search parameters
+     */
+    async findInitialRelatedWorks(query: string, contentFilter: models.ContentFilter): Promise<documents.WorkDocument[]> {
+        let findQuery = {'$text': {'$search': query}, 'audit.published': models.ApprovalStatus.Approved, 'audit.isDeleted': false};
+
+        switch (contentFilter) {
+            case models.ContentFilter.Everything: 
```

Comment:
Do we have an index on content filter by any chance? Might improve search performance for now.